### ISSUE TITLE
Use special cases internally for rotate & generating circles.

### ISSUE THREAD

**Issue by MichaelAtOz** [openscad/openscad#1602]
[Forum](http://forum.openscad.org/Bug-in-OpenScad-rendering-tp16396p16406.html)

"Yes but the OpenScad sin and cos functions exposed to the user has special cases for 0, 30, 45, 90, etc. Unfortunately it doesn't use those internally for rotate and generating circles."

Use special cases internally for rotate & generating circles.

**Comment by nophead** [comment]
Yes actually everywhere sin and cos or tan is used in OpenScad, which is rotate, circles, cylinders, spheres, rotate_extrude, offset, DXF import, off the top of my head. That would allow vertices to line up better.
